mscore/preferences.h

```diff
@@ -233,21 +233,52 @@ extern Preferences preferences;
 
 // Stream operators for enum classes
 // enum classes don't play well with QSettings without custom serialization
-template<typename T, typename std::enable_if<std::is_enum<T>::value>::type* = nullptr>
-inline QDataStream &operator<<(QDataStream &out, const T &val)
+inline QDataStream&
+operator<<(QDataStream &out, const Ms::MuseScoreStyleType &val)
 {
     return out << static_cast<int>(val);
 }
 
-template<typename T, typename std::enable_if<std::is_enum<T>::value>::type* = nullptr>
-inline QDataStream &operator>>(QDataStream &in, T &val)
```

Comment:
Did these templates prevent MuseScore from compiling? Could you please post the error message then?

Reply:
See https://musescore.org/en/node/299162

Reply:
So this seems to be because of these lines which were added in Qt 5.14:
https://github.com/qt/qtbase/blob/5.14.0/src/corelib/serialization/qdatastream.h#L387-L395

Actually Qt implementation of these operators seems to do the same but maybe it is indeed better to be explicit here.

Reply:
Yes, Qt implementation of these operators seems to do the same but actually differs: the build fails with it. This would need investigation but changes would be more involved.